frontend/src/metabase/query_builder/components/QueryDownloadWidget.jsx

```diff
@@ -162,17 +186,37 @@ const DashboardEmbedQueryButton = ({
   token,
   card,
   params,
+  setStatus,
 }) => (
   <DownloadButton
     method="GET"
     url={`api/embed/dashboard/${token}/dashcard/${dashcardId}/card/${card.id}/${type}`}
     extensions={[type]}
     params={params}
+    setStatus={setStatus}
   >
     {type}
   </DownloadButton>
 );
 
+const renderIcon = ({ icon, status }) => {
+  if ([`idle`, `resolved`, `rejected`].includes(status)) {
+    return (
+      <Tooltip tooltip={t`Download full results`}>
+        <Icon title={t`Download this data`} name={icon} size={20} />
+      </Tooltip>
+    );
+  } else if (status === "pending") {
+    return (
+      <Tooltip tooltip={t`Downloading...`}>
+        <LoadingSpinner />
```

Comment:
The spinner just looks a bit big right now. This little tweak makes it look more like it fits in with the other icons (screenshot below).

```suggestion
        <LoadingSpinner size={18} />
```

<img width="302" alt="image" src="https://user-images.githubusercontent.com/2223916/158277914-c9099402-5533-4588-8c4d-c548b0ac6548.png">